crates/astria-core/src/protocol/transaction/v1alpha1/action_group.rs

```diff
@@ -0,0 +1,491 @@
+        SudoAddressChangeAction,
+        TransferAction,
+        ValidatorUpdate,
+    },
+    Action,
+};
+
+trait BelongsToGroup {
+    const GROUP: ActionGroup;
+}
+
+macro_rules! impl_belong_to_group {
+    ($(($act:ty, $group:expr)),*$(,)?) => {
+        $(
+            impl BelongsToGroup for $act {
+                const GROUP: ActionGroup = $group;
+            }
+        )*
+    }
+}
+
+impl_belong_to_group!(
+    (
+        SequenceAction,
+        ActionGroup::BundlableGeneral(BundlableGeneral)
+    ),
+    (
+        TransferAction,
+        ActionGroup::BundlableGeneral(BundlableGeneral)
+    ),
+    (
+        ValidatorUpdate,
+        ActionGroup::BundlableGeneral(BundlableGeneral)
+    ),
+    (SudoAddressChangeAction, ActionGroup::Sudo(Sudo)),
+    (
+        IbcRelayerChangeAction,
+        ActionGroup::BundlableSudo(BundlableSudo)
+    ),
+    (
+        Ics20Withdrawal,
+        ActionGroup::BundlableGeneral(BundlableGeneral)
+    ),
+    (InitBridgeAccountAction, ActionGroup::General(General)),
+    (
+        BridgeLockAction,
+        ActionGroup::BundlableGeneral(BundlableGeneral)
+    ),
+    (
+        BridgeUnlockAction,
+        ActionGroup::BundlableGeneral(BundlableGeneral)
+    ),
+    (BridgeSudoChangeAction, ActionGroup::General(General)),
+    (FeeChangeAction, ActionGroup::BundlableSudo(BundlableSudo)),
+    (
+        FeeAssetChangeAction,
+        ActionGroup::BundlableSudo(BundlableSudo)
+    ),
+    (IbcRelay, ActionGroup::BundlableGeneral(BundlableGeneral)),
+);
+
+pub trait Group {
+    fn group(&self) -> ActionGroup;
+}
+
+impl Group for Action {
+    fn group(&self) -> ActionGroup {
+        match self {
+            Action::Sequence(_) => SequenceAction::GROUP,
+            Action::Transfer(_) => TransferAction::GROUP,
+            Action::ValidatorUpdate(_) => ValidatorUpdate::GROUP,
+            Action::SudoAddressChange(_) => SudoAddressChangeAction::GROUP,
+            Action::IbcRelayerChange(_) => IbcRelayerChangeAction::GROUP,
+            Action::Ics20Withdrawal(_) => Ics20Withdrawal::GROUP,
+            Action::InitBridgeAccount(_) => InitBridgeAccountAction::GROUP,
+            Action::BridgeLock(_) => BridgeLockAction::GROUP,
+            Action::BridgeUnlock(_) => BridgeUnlockAction::GROUP,
+            Action::BridgeSudoChange(_) => BridgeSudoChangeAction::GROUP,
+            Action::FeeChange(_) => FeeChangeAction::GROUP,
+            Action::FeeAssetChange(_) => FeeAssetChangeAction::GROUP,
+            Action::Ibc(_) => IbcRelay::GROUP,
+        }
+    }
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct BundlableGeneral;
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct General;
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct BundlableSudo;
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct Sudo;
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum ActionGroup {
+    BundlableGeneral(BundlableGeneral),
+    General(General),
```

Comment:
```suggestion
    UnbundleableGeneral(UnbundleableGeneral),
```

I think `General` is a bit unclear, would prefer to explicitly say unbundleable (and maybe even rename BundleableGeneral to just General?)

Reply:
I agree, I'll change the names to note the unbundleable instead!